pkg/providers/imagefamily/image_test.go

```diff
@@ -140,6 +140,31 @@ var _ = Describe("Image ID Resolution", func() {
 		Entry("Image version is empty, should get latest", imagefamily.Ubuntu2204Gen2CommunityImage, imagefamily.AKSUbuntuPublicGalleryURL, "", fmt.Sprintf("/CommunityGalleries/%s/images/%s/versions/%s", imagefamily.AKSUbuntuPublicGalleryURL, imagefamily.Ubuntu2204Gen2CommunityImage, latestImageVersion)),
 		Entry("Image version is specified, should use it", imagefamily.Ubuntu2204Gen2CommunityImage, imagefamily.AKSUbuntuPublicGalleryURL, olderImageVersion, fmt.Sprintf("/CommunityGalleries/%s/images/%s/versions/%s", imagefamily.AKSUbuntuPublicGalleryURL, imagefamily.Ubuntu2204Gen2CommunityImage, olderImageVersion)),
 	)
+	azLinuxNodeClass := test.AKSNodeClass()
+	azLinuxNodeClass.Spec.ImageFamily = lo.ToPtr(v1alpha2.AzureLinuxImageFamily)
+	DescribeTable("AzureLinux Image Resolution",
+		func(nodeClass *v1alpha2.AKSNodeClass, instanceType *cloudprovider.InstanceType, expectedImageID string) {
+			imageID, err := imageProvider.Get(context.Background(), nodeClass, instanceType, imagefamily.AzureLinux{})
```

Comment:
Can we start shifting more towards more coarse unit tests/ Acceptance tests, away from testing individual providers? An equivalent Acceptance test checking that, given the appropriate NodePool/AKSNodeClass and workload, a VM with AzureLinux image gets provisioned, would exercise the same provider code in the process.

(It may also bring forward the fact that we don't allow specifying HyperV generation externally any more ...)

Reply:
Yes we can and I did